libsolidity/formal/SMTEncoder.cpp

```diff
@@ -526,24 +529,21 @@ void SMTEncoder::endVisit(Identifier const& _identifier)
 	}
 	else if (_identifier.annotation().type->category() == Type::Category::Function)
 		visitFunctionIdentifier(_identifier);
-	else if (smt::isSupportedType(_identifier.annotation().type->category()))
+	else if (auto decl = identifierToVariable(_identifier))
+		defineExpr(_identifier, currentValue(*decl));
```

Comment:
Moved outside the supported if so that variables of unsupported types also have symbolic expressions.